mllib/src/test/scala/org/apache/spark/ml/fpm/FPGrowthSuite.scala

```diff
@@ -34,86 +35,122 @@ class FPGrowthSuite extends SparkFunSuite with MLlibTestSparkContext with Defaul
+            s"Expected $expected but found $prediction for data type $dt")
+        }(dt.encoder)
+      }
     }
   }
 
   test("FPGrowth getFreqItems") {
     val model = new FPGrowth().setMinSupport(0.7).fit(dataset)
-    val expectedFreq = spark.createDataFrame(Seq(
+    val expectedFreq = Seq(
       (Array("1"), 4L),
       (Array("2"), 3L),
       (Array("1", "2"), 3L),
       (Array("2", "1"), 3L) // duplicate as the items sequence is not guaranteed
-    )).toDF("items", "expectedFreq")
+    ).toDF("items", "expectedFreq")
     val freqItems = model.freqItemsets
 
     val checkDF = freqItems.join(expectedFreq, "items")
     assert(checkDF.count() == 3 && checkDF.filter(col("freq") === col("expectedFreq")).count() == 3)
   }
 
   test("FPGrowth getFreqItems with Null") {
-    val df = spark.createDataFrame(Seq(
+    val df = Seq(
       (1, Array("1", "2", "3", "5")),
       (2, Array("1", "2", "3", "4")),
       (3, null.asInstanceOf[Array[String]])
-    )).toDF("id", "items")
+    ).toDF("id", "items")
     val model = new FPGrowth().setMinSupport(0.7).fit(dataset)
-    val prediction = model.transform(df)
-    assert(prediction.select("prediction").where("id=3").first().getSeq[String](0).isEmpty)
+    testTransformerByGlobalCheckFunc[(Int, Array[String])](df, model, "id", "prediction") {
+      rows => {
+        val predictionForId3 = rows.filter(_.getAs[Int]("id") == 3)
+          .map(_.getAs[Seq[String]]("prediction"))
+        assert(Seq(Seq.empty) === predictionForId3,
+          s"Expected empty prediction for id 3, got $predictionForId3")
+      }
+    }
   }
 
   test("FPGrowth prediction should not contain duplicates") {
     // This should generate rule 1 -> 3, 2 -> 3
-    val dataset = spark.createDataFrame(Seq(
+    val dataset = Seq(
       Array("1", "3"),
       Array("2", "3")
-    ).map(Tuple1(_))).toDF("items")
+    ).toDF("items")
     val model = new FPGrowth().fit(dataset)
 
-    val prediction = model.transform(
-      spark.createDataFrame(Seq(Tuple1(Array("1", "2")))).toDF("items")
-    ).first().getAs[Seq[String]]("prediction")
+    val df = Seq(Array("1", "2")).toDF("items")
 
-    assert(prediction === Seq("3"))
+    testTransformerByGlobalCheckFunc[(Array[String])](df, model, "prediction") {
+      rows => {
+        assert(1 === rows.size, s"Expected exactly 1 prediction, got $rows")
+        val predictions = rows.map(_.getAs[Seq[String]]("prediction"))
+        val expected = Seq(Seq("3"))
+        assert(expected === predictions, s"Expected $expected, got $predictions")
+      }
+    }
   }
 
   test("FPGrowthModel setMinConfidence should affect rules generation and transform") {
     val model = new FPGrowth().setMinSupport(0.1).setMinConfidence(0.1).fit(dataset)
     val oldRulesNum = model.associationRules.count()
-    val oldPredict = model.transform(dataset)
+    val oldPredict = model.transform(dataset).withColumnRenamed("prediction", "oldPrediction")
```

Comment:
Other places named it `oldPrediction`, this could be also then renamed.

Reply:
`oldPrediction` is used only in this test case, in pattern matching. If I rename the variable it will be shadowed in the pattern match which is probably not what we want.